test/proxy/upgradeable/UpgradeableProxy.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from 'chai';
+import { ethers } from 'hardhat';
+import { deployMockContract } from 'ethereum-waffle';
+import { describeBehaviorOfUpgradeableProxy } from '@solidstate/spec';
+import {
+  UpgradeableProxyMock,
+  UpgradeableProxyMock__factory,
+} from '../../../typechain';
+
+describe('UpgradeableProxy', function () {
+  let instance: UpgradeableProxyMock;
+
+  beforeEach(async function () {
+    const implementationFactory = await ethers.getContractFactory(
+      'OwnableMock',
+    );
+    const implementationInstance = await implementationFactory.deploy(
+      ethers.constants.AddressZero,
+    );
+    await implementationInstance.deployed();
+
+    const [deployer] = await ethers.getSigners();
+    instance = await new UpgradeableProxyMock__factory(deployer).deploy(
+      implementationInstance.address,
+    );
+  });
+
+  describeBehaviorOfUpgradeableProxy({
+    deploy: async () => instance as any,
+    implementationFunction: 'owner()',
+    implementationFunctionArgs: [],
+  });
+
+  describe('__internal', function () {
+    describe('#_getImplementation', function () {
+      it('returns implementation address', async function () {
+        expect(await instance.callStatic.__getImplementation()).to.be
+          .properAddress;
```

Comment:
No need, because the `#_setImplementation` test validates that the correct address is returned.  Testing that here would really just be testing that the mock contract sets the right address (but we don't really care about the mock contract).